packages/contracts/contracts/registry/PluginRepoRegistry.sol

```diff
@@ -20,19 +24,25 @@ contract PluginRepoRegistry is InterfaceBasedRegistry {
 
     /// @notice Initializes the contract by setting calling the `InterfaceBasedRegistry` base class initialize method.
     /// @param _dao The address of the managing DAO.
-    function initialize(IDAO _dao) public initializer {
+    function initialize(IDAO _dao, ENSSubdomainRegistrar _subdomainRegistrar) public initializer {
         bytes4 pluginRepoInterfaceId = type(IPluginRepo).interfaceId;
         __InterfaceBasedRegistry_init(_dao, pluginRepoInterfaceId);
+
+        subdomainRegistrar = _subdomainRegistrar;
     }
 
     /// @notice Registers a plugin repository with a name and address.
     /// @param name The name of the PluginRepo.
     /// @param registrant The address of the PluginRepo contract.
-    function registerPlugin(string calldata name, address registrant)
+    function registerPluginRepo(string calldata name, address registrant)
         external
         auth(PLUGIN_REGISTER_PERMISSION_ID)
```

Comment:
We should rename the permission ID to match with the function name

`REGISTER_PLUGIN_REPO_PERMISSION_ID`